app/uk/gov/hmrc/estates/transformers/DeltaTransform.scala

```diff
@@ -17,10 +17,25 @@
 package uk.gov.hmrc.estates.transformers
 
 import play.api.libs.json.{JsValue, _}
-import uk.gov.hmrc.estates.transformers.register.{AgentDetailsTransform, AmountOfTaxOwedTransform, DeceasedTransform}
+import uk.gov.hmrc.estates.transformers.register.{AgentDetailsTransform, AmountOfTaxOwedTransform, DeceasedTransform, YearsReturnsTransform}
 
 trait DeltaTransform {
-  def applyTransform(input: JsValue): JsResult[JsValue]
+
+  val path: JsPath = __
+
+  val value: JsValue = Json.obj()
```

Comment:
It feels wrong to me to have this implementation detail in the trait. It means that all transforms would have it or need it. I see it's defined by default here, so you don't have the practical pain of having to define it further up. But it seems like it should be a trait (or abstract class) derived from this without defaults that classes that need it would use, without cluttering up the pure trait.

Reply:
Also, if it's a higher level abstract class, you could have a constructor for it that takes the two params, so you don't need to define them as separate overrides. (This might be me just not totally liking the whole override thing for values. It does seem to be a pattern, though, so it might just be me.
)